FsHealthServiceTests testFailsHealthOnIOException on Windows

**Build scan**:
https://gradle-enterprise.elastic.co/s/schg2cx7kb3ja
https://gradle-enterprise.elastic.co/s/avmtx56geouvy
https://gradle-enterprise.elastic.co/s/kkawj75b45nzc

**Repro line**:
```
 gradlew ':server:test' --tests "org.elasticsearch.monitor.fs.FsHealthServiceTests.testFailsHealthOnIOException" \
  -Dtests.seed=383792DA00F947A9 \
  -Dtests.security.manager=true \
  -Dtests.locale=sv-SE \
  -Dtests.timezone=AGT \
  -Druntime.java=11
```

**Reproduces locally?**:
Unknown, Windows innit 

**Applicable branches**:
Master, 7.x

**Failure history**:
Regularly

https://build-stats.elastic.co/app/kibana#/discover?_g=(refreshInterval:(pause:!t,value:0),time:(from:now-7d,mode:quick,to:now))&_a=(columns:!(branch,build_url,build.branch,build.console-url),index:b646ed00-7efc-11e8-bf69-63c8ef516157,interval:auto,query:(language:lucene,query:'FsHealthServiceTests%20testFailsHealthOnIOException'),sort:!(process.time-start,desc))


**Failure excerpt**:
```
java.lang.AssertionError: expected:<UNHEALTHY> but was:<HEALTHY>
	at __randomizedtesting.SeedInfo.seed([98825EFCB7B3E2A5:FC2CB339E1C9370]:0)
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.failNotEquals(Assert.java:834)
	at org.junit.Assert.assertEquals(Assert.java:118)
	at org.junit.Assert.assertEquals(Assert.java:144)
	at org.elasticsearch.monitor.fs.FsHealthServiceTests.testFailsHealthOnIOException(FsHealthServiceTests.java:119)
```

--- a/server/src/test/java/org/elasticsearch/monitor/fs/FsHealthServiceTests.java
+++ b/server/src/test/java/org/elasticsearch/monitor/fs/FsHealthServiceTests.java
@@ -100,8 +100,6 @@ public class FsHealthServiceTests extends ESTestCase {
     }
 
     public void testFailsHealthOnIOException() throws IOException {
-        assumeFalse("https://github.com/elastic/elasticsearch/issues/59380", Constants.WINDOWS);
-        
         FileSystem fileSystem = PathUtils.getDefaultFileSystem();
         FileSystemIOExceptionProvider disruptFileSystemProvider = new FileSystemIOExceptionProvider(fileSystem);
         fileSystem = disruptFileSystemProvider.getFileSystem(null);
@@ -116,6 +114,7 @@ public class FsHealthServiceTests extends ESTestCase {
             assertEquals("health check passed", fsHealthService.getHealth().getInfo());
 
             //disrupt file system
+            disruptFileSystemProvider.restrictPathPrefix(""); // disrupt all paths
             disruptFileSystemProvider.injectIOException.set(true);
             fsHealthService = new FsHealthService(settings, clusterSettings, testThreadPool, env);
             fsHealthService.new FsHealthMonitor().run();
@@ -221,9 +220,9 @@ public class FsHealthServiceTests extends ESTestCase {
             assertEquals("health check passed", fsHealthService.getHealth().getInfo());
 
             //disrupt file system writes on single path
-            disruptWritesFileSystemProvider.injectIOException.set(true);
             String disruptedPath = randomFrom(paths).toString();
             disruptWritesFileSystemProvider.restrictPathPrefix(disruptedPath);
+            disruptWritesFileSystemProvider.injectIOException.set(true);
             fsHealthService = new FsHealthService(settings, clusterSettings, testThreadPool, env);
             fsHealthService.new FsHealthMonitor().run();
             assertEquals(UNHEALTHY, fsHealthService.getHealth().getStatus());
@@ -241,7 +240,7 @@ public class FsHealthServiceTests extends ESTestCase {
         AtomicBoolean injectIOException = new AtomicBoolean();
         AtomicInteger injectedPaths = new AtomicInteger();
 
-        private String pathPrefix = "/";
+        private String pathPrefix;
 
         FileSystemIOExceptionProvider(FileSystem inner) {
             super("disrupt_fs_health://", inner);
@@ -258,6 +257,7 @@ public class FsHealthServiceTests extends ESTestCase {
         @Override
         public OutputStream newOutputStream(Path path, OpenOption... options) throws IOException {
             if (injectIOException.get()){
+                assert pathPrefix != null : "must set pathPrefix before starting disruptions";
                 if (path.toString().startsWith(pathPrefix) && path.toString().endsWith(".es_temp_file")) {
                     injectedPaths.incrementAndGet();
                     throw new IOException("fake IOException");
@@ -272,7 +272,7 @@ public class FsHealthServiceTests extends ESTestCase {
         AtomicBoolean injectIOException = new AtomicBoolean();
         AtomicInteger injectedPaths = new AtomicInteger();
 
-        private String pathPrefix = "/";
+        private String pathPrefix = null;
 
         FileSystemFsyncIOExceptionProvider(FileSystem inner) {
             super("disrupt_fs_health://", inner);
@@ -292,6 +292,7 @@ public class FsHealthServiceTests extends ESTestCase {
                 @Override
                 public void force(boolean metaData) throws IOException {
                     if (injectIOException.get()) {
+                        assert pathPrefix != null : "must set pathPrefix before starting disruptions";
                         if (path.toString().startsWith(pathPrefix) && path.toString().endsWith(".es_temp_file")) {
                             injectedPaths.incrementAndGet();
                             throw new IOException("fake IOException");

--- a/server/src/test/java/org/elasticsearch/monitor/fs/FsHealthServiceTests.java
+++ b/server/src/test/java/org/elasticsearch/monitor/fs/FsHealthServiceTests.java
@@ -17,16 +17,13 @@
  * under the License.
  */
 
-
 package org.elasticsearch.monitor.fs;
-
 
 import org.apache.logging.log4j.Level;
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 import org.apache.lucene.mockfile.FilterFileChannel;
 import org.apache.lucene.mockfile.FilterFileSystemProvider;
-import org.apache.lucene.util.Constants;
 import org.elasticsearch.cluster.coordination.DeterministicTaskQueue;
 import org.elasticsearch.common.io.PathUtils;
 import org.elasticsearch.common.io.PathUtilsForTesting;
@@ -57,7 +54,6 @@ import static org.elasticsearch.monitor.StatusInfo.Status.HEALTHY;
 import static org.elasticsearch.monitor.StatusInfo.Status.UNHEALTHY;
 import static org.elasticsearch.node.Node.NODE_NAME_SETTING;
 import static org.hamcrest.Matchers.is;
-
 
 public class FsHealthServiceTests extends ESTestCase {
 

